sunpy/net/hek/utils.py

```diff
@@ -0,0 +1,281 @@
+            if coord3:
+                event_coord = SkyCoord(coord1*unit, coord2*unit, coord3*unit, frame=frame)
+            else:
+                event_coord = SkyCoord(coord1*unit, coord2*unit, frame=frame)
+            event_coord_col.append(event_coord)
+        event_coord_col = Column(event_coord_col, name='event_coord')
+        del table['event_coord1']
+        del table['event_coord2']
+        del table['event_coord3']
+        table.add_column(event_coord_col)
+
+    for attribute in attributes:
+        if attribute["name"] in table.colnames and ("unit_prop" in attribute or attribute.get("is_chaincode", False)) and attribute.get("is_unit_prop", True):
+            unit_attr = ""
+            if is_coord_prop:
+                unit_attr = "event_coordunit"
+            else:
+                unit_attr = attribute["unit_prop"]
+
+            new_column = []
+            for idx, value in enumerate(table[attribute["name"]]):
+                new_value = ""
+                if value in ["", None]:
+                    new_value = value
+                elif attribute.get("is_chaincode", False):
+                    unit = table[attribute["unit_prop"]][idx]
+                    time = table['event_starttime'][idx]
+                    new_value = parse_chaincode(value, attribute, unit, time)
+                else:
+                    unit = get_unit(table[unit_attr][idx])
+                    new_value = value * unit
+                    if attribute["name"] == 'obs_meanwavel':
+                        new_value = new_value.to_value(u.angstrom)
+                new_column.append(new_value)
+
+            if not attribute.get("is_chaincode"):
+                new_column = Column(new_column, name=table[attribute["name"]], dtype=u.Quantity)
+            table[attribute["name"]] = new_column
+
+    for attribute in attributes:
+        if attribute.get("is_unit_prop", False) and attribute["name"] in table.colnames:
+            del table[attribute["name"]]
+    return table
+
+def parse_chaincode(value, attribute, unit, time):
+    """
+    Parses a string representation of coordinates and convert them into a PolygonSkyRegion object
+    using units based on the specified coordinate frame.
+
+    Parameters
+    ----------
+    value : str
+        A polygon defined using vertices in sky coordinates.
+    attribute : dict
+        An object from "coord_properties.json"
+    unit : str
+        The unit of the coordinates
+    time : `~astropy.time.core.Time`
+        An event_starttime row parsed into astropy time.
+
+    Returns
+    -------
+    `PolygonSkyRegion`
+        A polygon defined using vertices in sky coordinates.
+
+    Raises
+    ------
+    IndexError
+        Because ``value`` does not contain the expected '((' and '))' substrings.
+    UnitConversionError
+        Because the units set by ``coord1_unit`` or ``coord2_unit`` are incompatible with the values being assigned.
+    """
+    observer = 'earth'
```

Comment:
Setting `observer = 'earth'` is an assumption, right? Maybe a comment should be added to make this explicit?